src/shared/sleep-config.c

```diff
@@ -207,67 +292,203 @@ static bool battery_discharge_rate_is_valid(int battery_discharge_rate) {
+                if (r != 2)
+                        return log_debug_errno(SYNTHETIC_ERRNO(EINVAL), "Invalid number of items fetched from %s", DISCHARGE_RATE_FILEPATH);
 
-        r = safe_atoi(stored_discharge_rate, &discharge_rate);
-        if (r < 0)
-               return log_debug_errno(r, "Failed to parse discharge rate read from %s location: %m", DISCHARGE_RATE_FILEPATH);
+                r = safe_atou64(stored_hash_id, &hash_id);
+                if (r < 0)
+                        return log_debug_errno(r, "Failed to parse discharge rate read from %s location: %m", DISCHARGE_RATE_FILEPATH);
 
-        if (!battery_discharge_rate_is_valid(discharge_rate))
-               return log_debug_errno(SYNTHETIC_ERRNO(ERANGE), "Invalid battery discharge percentage rate per hour: %m");
+                r = get_system_battery_identifier_hash(dev, &current_hash_id);
+                if (r < 0)
+                        return log_debug_errno(r, "Failed to generate system battery identifier hash: %m");
+
+                if (current_hash_id != hash_id)
+                /* matching device not found, move to next line */
+                        continue;
+
+                r = safe_atoi(stored_discharge_rate, &discharge_rate);
+                if (r < 0)
+                        return log_debug_errno(r, "Failed to parse discharge rate read from %s location: %m", DISCHARGE_RATE_FILEPATH);
+
+                if (!battery_discharge_rate_is_valid(discharge_rate))
+                        return log_debug_errno(SYNTHETIC_ERRNO(ERANGE), "Invalid battery discharge percentage rate per hour: %m");
 
-        return discharge_rate;
+                *ret = discharge_rate;
+                break; /* matching device found, exit iteration */
+        }
+
+        return 0;
 }
 
 /* Write battery percentage discharge rate per hour along with system and battery identifier hash to file */
-int put_battery_discharge_rate(int estimated_battery_discharge_rate) {
-        uint64_t system_hash_id;
+static int put_battery_discharge_rate(int estimated_battery_discharge_rate, uint64_t system_hash_id, bool truncate) {
         int r;
 
+        assert(system_hash_id);
+        assert(truncate);
+
         if (!battery_discharge_rate_is_valid(estimated_battery_discharge_rate))
                return log_debug_errno(SYNTHETIC_ERRNO(ERANGE), "Invalid battery discharge percentage rate per hour: %m");
 
-        r = get_system_battery_identifier_hash(&system_hash_id);
-        if (r < 0)
-               return log_debug_errno(r, "Failed to generate system battery identifier hash: %m");
-
-        r = write_string_filef(
+        if (truncate)
+                r = write_string_filef(
                         DISCHARGE_RATE_FILEPATH,
-                        WRITE_STRING_FILE_CREATE|WRITE_STRING_FILE_MKDIR_0755,
+                        WRITE_STRING_FILE_CREATE | WRITE_STRING_FILE_MKDIR_0755 | WRITE_STRING_FILE_TRUNCATE,
```

Comment:
instead of an if, simply do:

`WRITE_STRING_FILE_CREATE | WRITE_STRING_FILE_MKDIR_0755 | (truncate ? WRITE_STRING_FILE_TRUNCATE : 0)`

Reply:
yeah nice and short